docs/source/data.rst

```diff
@@ -2,11 +2,342 @@ torch.utils.data
 ===================================
 
 .. automodule:: torch.utils.data
+
+At the heart of PyTorch data loading utility is the :class:`torch.utils.data.DataLoader`
+class.  It represents an Python Iterable over a dataset, with
+`the iterating strategy <Data Loading Strategies_>`_ specified by
+`the type of the given dataset <Dataset Types_>`_  and the constructor arguments.
+Moreover, it supports
+`both single- and multi-process data loading <Single- and Multi-process Data Loading_>`_,
+as well as automatic `memory pinning <Memory Pinning_>`_.
+
+Dataset Types
+-------------
+
+:class:`~torch.utils.data.DataLoader` supports two different types of datasets:
+
+* `map-style datasets <Map-style datasets_>`_
+
+* `iterable-style datasets <Iterable-style datasets_>`_
+
+Map-style datasets
+^^^^^^^^^^^^^^^^^^
+
+A map-style dataset is one that implements the ``__getitem__`` protocol,
```

Comment:
It would be nice to mention it as `__len__` + `__getitem__` because you really need both of those.